cdm/src/main/java/ucar/nc2/NetcdfFile.java

```diff
@@ -693,9 +701,12 @@ static private String makeUncompressed(String filename) throws Exception {
 
       } finally {
         if (lock != null) lock.release();
+        stringKeeper.release(filename);
       }
     }
 
+    stringKeeper.release(filename);
```

Comment:
Better in a single try finally block so we always know its going to be released?

Reply:
I have done so in the second commit.  There was already a try..catch in the method that calls makeUncompressed (getRaf) which I added the finally to.